plugins/api/README.md

```diff
@@ -111,6 +115,36 @@ data class ExampleConfiguration(
 
 Here, the `serverUrl` property has a default value, the `timeout` property is required, and the `token` property is optional.
 
+The generated `ExampleAdvisorFactory` class will contain two factory functions:
+
+```kotlin
+override fun create(config: PluginConfig): ExampleAdvisor {
+  val configObject = ExampleConfiguration(
+      serverUrl = config.options["serverUrl"] ?: "https://example.com",
+      timeout = config.options["timeout"]?.toInt(),
+      token = config.secrets["token"]
+  )
+
+  return ExampleAdvisor(descriptor, configObject)
+}
+
+companion object {
```

Comment:
I assume this is not really a top-level companion object, but you omitted the surrounding `ExampleAdvisor`. Should we still add it here to avoid the question why this is not a top-level function?